src/containerapp/azext_containerapp/_utils.py

```diff
@@ -788,6 +788,35 @@ def _remove_readonly_attributes(containerapp_def):
             del containerapp_def['properties'][unneeded_property]
 
 
+def clean_null_values(d):
```

Comment:
I believe we're doing this because null values in the PATCH payload make the API delete those properties. The API expects values to not be in the payload at all if we don't want to update them. The SDK takes care of removing null values for us, but for now we have to do that ourselves. I'll add a comment for this 

@haroonf correct me that's incorrect